### PR TITLE
[PIR] register some fused_ops

### PR DESCRIPTION
**Pull request by kangguangli**
### PR types

Others
### PR changes

Others
### Description

1. register `fused_attention_grad` in pir
1. register `fused_feedforward` and its grad op in pir
2. migrate `FusedAttentionInfermeta` from `multiary.cc` to `fusion.cc` by comments in #57557 
3. delete some redundant unittests
#### Others
Pcard-67164

### CODE REVIEW

File: paddle/fluid/ir_adaptor/translator/op_translator.cc [hunk 1]
```diff
@@ -1808,6 +1808,65 @@ struct LegacySetValueDispatcher : public OpTranscriber {
...
+  }
+
+  void RecordOpResultMapping(pir::IrContext* ctx,
+                             TranslationContext* param_map,
+                             const OpDesc& op_desc,
+                             pir::Operation* operation,
+                             const OpOutputMapping& arg_to_idx) override {
+    OpTranscriber::RecordOpResultMapping(
+        ctx, param_map, op_desc, operation, arg_to_idx);
+    if (op_desc.HasOutput("Out")) {
```
Comment (kangguangli) [inline paddle/fluid/ir_adaptor/translator/op_translator.cc:1855]: Sometimes, the original `fused_feedforward` will have two outputs pointing to same variable, which uses same memory buffer. 
https://github.com/PaddlePaddle/Paddle/blob/3ce66d58c87c221e5b56af624f9f06e1be58fd35/paddle/fluid/framework/ir/fused_feedforward_pass.cc#L188-L192
After translating, the two outputs will be mapped to two values while having different memory buffers and result in precision problem.

By forcing this duplicate variable mapping to  output `out` only, we can get the right result.

**Review by XieYunshen — DISMISSED** [review]
LGTM 
`单测删除`

File: paddle/phi/api/yaml/op_compat.yaml [hunk 2]
```diff
@@ -1254,6 +1255,38 @@
     attrs : [bool use_cudnn = false, float fuse_alpha = 0.0f, float fuse_beta = 0.0f, float Scale_in = 1.0f,
              float Scale_out = 1.0f, float Scale_in_eltwise = 1.0f, 'float[] Scale_weights = {1.0f}']
 
+- op : fused_feedforward
+  backward: fused_feedforward_grad
+  inputs:
+    x: X
+    dropout1_seed: Dropout1Seed
+    dropout2_seed: Dropout2Seed
```
Comment (zyfncg) [inline paddle/phi/api/yaml/op_compat.yaml:1263]: 这两个是不可变attribute？

**Review by zyfncg — DISMISSED** [review]

File: paddle/phi/api/yaml/op_compat.yaml (hunk 2 above)
Comment (kangguangli) [reply paddle/phi/api/yaml/op_compat.yaml:1263]: 不是，fused_feedforward的kernel参数重既包含dropout1_seed，又包含dropout1_seed_val

File: paddle/fluid/ir_adaptor/translator/op_translator.cc (hunk 1 above)
Comment (zhangbo9674) [reply paddle/fluid/ir_adaptor/translator/op_translator.cc:1855]: 建议针对该问题添加必要 Warning 日志

**Review by XieYunshen — APPROVED** [review]

**Review by zyfncg — APPROVED** [review]

**Pull request merged**